This patch can be removed once the issue referenced below is resolved.

diff -uNr hledger-iadd-1.3.20.orig/cabal.project hledger-iadd-1.3.20/cabal.project
--- hledger-iadd-1.3.20.orig/cabal.project	1970-01-01 01:00:00.000000000 +0100
+++ hledger-iadd-1.3.20/cabal.project	2024-04-08 19:32:40.738079452 +0200
@@ -0,0 +1,7 @@
+packages: *.cabal
+
+-- The most recent hledger release doesn't offically support GHC 9.X yet.
+-- We can work around that by allowing a newer base version for hledger-lib.
+--
+-- See: https://github.com/simonmichael/hledger/issues/2149
+allow-newer: hledger-lib, hledger-lib:base
